draft-sharabayko-mops-srt.md

```diff
@@ -2086,15 +2086,584 @@ and RTTVar are updated by the socket based on algorithms for the sender
+avoidance phase and behaves as described below.
+
+On an acknowledgement (ACK) packet reception:
+
+Step 1. If the time passed since the last time LastRCTime the sending rate
+was either increased, or kept, is less than RC_INTERVAL:
+
+a. Keep the sending rate at the same level;
+
+b. Stop.
+
+    <CODE BEGINS>
+    if (currTime - LastRCTime < RC_INTERVAL)
+    {
+        Keep the sending rate at the same level;
+        Stop;
+    }
+    <CODE ENDS>
+
+where currTime is the current time, in microseconds; LastRCTime is
+the last time the sending rate was either increased, or kept,
+in microseconds.
+
+Step 2. Update the value of LastRCTime to the current time:
+
+~~~
+LastRCTime = currTime
+~~~
+
+Step 3. Set the congestion window size to:
+
+~~~
+CWND_SIZE = RECEIVING_RATE * (RTT + RC_INTERVAL) / 1000000 + 16
+~~~
+
+Step 4. If there is packet loss reported by the receiver (bLoss=True):
+
+a. Keep the value of PKT_SND_PERIOD at the same level;
+
+b. Set the value of bLoss to False;
+
+c. Stop.
+
+bLoss flag is equal to True if a packet loss has happened since the
+last sending rate increase. Initial value: False.
+
+Step 5. If there is no packet loss reported by the receiver (bLoss=False),
+calculate PKT_SND_PERIOD as follows:
 
-Given that SRT can operate in live and file transfer modes, there are two groups of congestion control algorithms possible.
+    <CODE BEGINS>
+    inc = 0;
+
+    lossBandwidth = 2 * (1000000 / LastDecPeriod);
+    linkCapacity = min(lossBandwidth, EST_LINK_CAPACITY);
+    B = linkCapacity - 1000000 / PKT_SND_PERIOD;
+
+    if ((PKT_SND_PERIOD > LastDecPeriod) && ((linkCapacity / 9) < B))
+        B = linkCapacity / 9;
+    if (B <= 0)
+        inc = 1 / S;
+    else
+    {
+        inc = pow(10.0, ceil(log10(B * S * 8))) * 0.0000015 / S;
+        inc = max(inc, 1 / S);
+    }
+
+    PKT_SND_PERIOD = (PKT_SND_PERIOD * RC_INTERVAL) / 
+                      (PKT_SND_PERIOD * inc + RC_INTERVAL);
+    <CODE ENDS>
+
+where
+
+- LastDecPeriod is the value of PKT_SND_PERIOD right before the last
+sending rate decrease has happened (on a loss report (NAK) packet
+reception), in microseconds. The initial value of LastDecPeriod
+is set to 1 microsecond;
+- EST_LINK_CAPACITY is the estimated link capacity reported
+by the receiver within an ACK packet and smoothed at the sender side
+({{bandwidth-estimation}), in packets per second;
+- B is the estimated available bandwidth, in packets per second;
+- S is the SRT packet size (in terms of IP payload) in bytes.
+SRT treats 1500 bytes as a standard packet size.
+
+The detailed explanation of the formulas used to calculate the increase
+in sending rate inc can be found in {{GuAnAO}}. The UDT's available
+bandwidth B estimation has been modified to take into account the
+bandwidth registered at the moment of packet loss since the estimated
+link capacity EST_LINK_CAPACITY reported by the receiver may
+overestimate the real one significantly.
```

Comment:
```suggestion
A detailed explanation of the formulas used to calculate the increase
in sending rate can be found in {{GuAnAO}}. UDT's available
bandwidth estimation has been modified to take into account the
bandwidth registered at the moment of packet loss, since the estimated
link capacity EST_LINK_CAPACITY reported by the receiver may
overestimate the actual link capacity significantly.
```